frontend/src/utils/jsonUtils.ts

```diff
@@ -0,0 +1,87 @@
+export const readFileAsText = (file: File): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+    reader.onload = () => resolve(reader.result as string);
+    reader.onerror = () =>
+      reject(new Error(`Failed to read file: ${file.name}`));
+    reader.readAsText(file);
+  });
+};
+
+export const validateJsonContent = (content: string): string | null => {
+  try {
+    JSON.parse(content);
+    return null;
+  } catch {
+    // If not regular JSON, check if it's newline-delimited JSON
+    return validateNewlineDelimitedJson(content);
```

Comment:
Will this catch block still produce valid error messages if it is not a newline delimited json and just a malformed normal json?